### PR TITLE
Increasing code coverage

### PR DESCRIPTION
**Pull request by ahcorde**
Increased the code coverage.

 - Added failing allocators
 - Untested methods
   - get_readers_info_by_topic
   - get_writers_info_by_topic
 - Operators
   - << 
   - ==

Signed-off-by: ahcorde <ahcorde@gmail.com>

### CODE REVIEW

File: rmw_dds_common/test/test_graph_cache.cpp [hunk 1]
```diff
@@ -1127,3 +1128,259 @@ TEST(test_graph_cache, normal_usage)
...
+    rmw_topic_endpoint_info_array_t topic_endpoint_info_array_sub =
+      rmw_get_zero_initialized_topic_endpoint_info_array();
+    rcutils_allocator_t allocator = rcutils_get_default_allocator();
+    rmw_ret_t ret = graph_cache.get_writers_info_by_topic(
+      "topic1",
+      identity_demangle,
+      &allocator,
+      &topic_endpoint_info_array_sub);
+    EXPECT_EQ(ret, RMW_RET_OK);
+  }
```
Comment (Blast545) [inline rmw_dds_common/test/test_graph_cache.cpp:1324]: Would it be possible/reasonable to check here the output of  the function, written into `topic_endpoint_info_array_sub`?

File: rmw_dds_common/test/test_graph_cache.cpp [hunk 2]
```diff
@@ -1127,3 +1128,259 @@ TEST(test_graph_cache, normal_usage)
...
+    rmw_topic_endpoint_info_array_t topic_endpoint_info_array_sub =
+      rmw_get_zero_initialized_topic_endpoint_info_array();
+    rcutils_allocator_t allocator = rcutils_get_default_allocator();
+    rmw_ret_t ret = graph_cache.get_readers_info_by_topic(
+      "topic1",
+      identity_demangle,
+      &allocator,
+      &topic_endpoint_info_array_sub);
+    EXPECT_EQ(ret, RMW_RET_OK);
+    rcutils_reset_error();
```
Comment (Blast545) [inline rmw_dds_common/test/test_graph_cache.cpp:1384]: As there is a failed allocator test for `get_writers_info_by_topic` sounds appropriate to include as well a failing allocator one for `get_readers_info_by_topic` as well

File: rmw_dds_common/test/test_graph_cache.cpp [hunk 3]
```diff
@@ -1127,3 +1128,259 @@ TEST(test_graph_cache, normal_usage)
...
+    "      namespace: 'ns2' name: 'node1'\n"
+    "      associated data readers gids:\n"
+    "      associated data writers gids:\n"
+    "---------------------------------\n";
+
+  // Add some nodes.
+  check_participant_entities_msg(
+    add_nodes(
+      graph_cache, {
+    {"participant1", "ns1", "node1"},
```
[the remark this answers is not in the document]
Comment (ahcorde) [reply rmw_dds_common/test/test_graph_cache.cpp:1176]: Using the same indentation as above. Linters are okay

File: rmw_dds_common/test/test_graph_cache.cpp [hunk 4]
```diff
@@ -1127,3 +1128,259 @@ TEST(test_graph_cache, normal_usage)
...
+    "      associated data writers gids:\n"
+    "        77.72.69.74.65.72.31.0.0.0.0.0.0.0.0.0.0.0.0.0.0.0.0.0\n"
+    "      namespace: 'ns2' name: 'node1'\n"
+    "      associated data readers gids:\n"
+    "      associated data writers gids:\n"
+    "---------------------------------\n";
+
+  add_entities(
+    graph_cache,
+  {
```
[the remark this answers is not in the document]
Comment (ahcorde) [reply rmw_dds_common/test/test_graph_cache.cpp:1218]: Using the same indentation as above. Linters are okay

File: rmw_dds_common/test/test_graph_cache.cpp (hunk 2 above)
Comment (ahcorde) [reply rmw_dds_common/test/test_graph_cache.cpp:1404]: https://github.com/ros2/rmw_dds_common/pull/14/commits/9f4a0f5269cc7a5e6e85f1356d236d00cd60c5c9

File: rmw_dds_common/test/test_graph_cache.cpp (hunk 1 above)
Comment (ahcorde) [reply rmw_dds_common/test/test_graph_cache.cpp:1319]: https://github.com/ros2/rmw_dds_common/pull/14/commits/9f4a0f5269cc7a5e6e85f1356d236d00cd60c5c9

**Comment by ahcorde** [comment]
* Linux [Build Status](http://ci.ros2.org/job/ci_linux/10309/)
* Linux-aarch64 [Build Status](http://ci.ros2.org/job/ci_linux-aarch64/5840/)
* macOS [Build Status](http://ci.ros2.org/job/ci_osx/8396/)
* Windows [Build Status](http://ci.ros2.org/job/ci_windows/10216/)

**Pull request merged**